CHANGELOG.md

```diff
@@ -18,6 +20,224 @@
+  valid JSON but neither an object nor an array [#4063](https://github.com/Kong/kong/pull/4063)
+
+A summary of the changes introduced by the New Admin API:
+
+- Pagination has been included in all "multi-record" endpoints, and pagination
+  control fields are different than in 0.14.x
+- Filtering now happens via URL path changes (`/consumers/x/plugins`) instead
+  of querystring fields (`/plugins?consumer_id=x`)
+- Array values can't be coherced from comma-separated strings. They must be
+  "proper" JSON values on JSON requests, or use a new syntax on form-url-encoded
+  or multipart requests.
+- Error messages have been been reworked from the ground up to be more
+  consistent, precise and informative.
+- The `PUT` method has been reimplemented with idempotent behavior and has
+  been added to some entities that didn't have it.
+
+For more details about the New Admin API, please visit the official docs:
+https://docs.konghq.com/
+
+##### Plugins
+
+- All bundled Plugins' schemas and custom entities have been updated to the new
+  `kong.db`, and their APIs have been updated to the New Admin API.
+  [#3766](https://github.com/Kong/kong/pull/3766),
+  [#3774](https://github.com/Kong/kong/pull/3774),
+  [#3778](https://github.com/Kong/kong/pull/3778),
+  [#3839](https://github.com/Kong/kong/pull/3839)
+- All plugin migrations have been converted to the new migration framework
+
+### Additions
+
+##### :fireworks: Service Mesh and Stream Routes
+
+- These new features require a patched version of OpenResty, but Kong will still
+  work fine on a non-patched OpenResty for HTTP(S) API Gateway scenarios
+- Support for TCP & TLS Stream Routes via the `stream_listen` config option
+  [#4009](https://github.com/Kong/kong/pull/4009)
+- New `origins` config property allows overriding hosts from kong [#3679](https://github.com/Kong/kong/pull/3679)
+- Kong instances can now create a shared internal Certificate Authority, which is
+  used for Service Mesh SSL traffic [#3906](https://github.com/Kong/kong/pull/3906), [#3861](https://github.com/Kong/kong/pull/3861),
+- `transparent` listeners allow setting up a Service Mesh with `iptables` [#3884](https://github.com/Kong/kong/pull/3884)
+- Plugins' `run_on` field to control how they behave on a Service Mesh
+  environment [#3930](https://github.com/Kong/kong/pull/3930), [#4066](https://github.com/Kong/kong/pull/4066)
+- There is a new phase called `preread`. This is where stream traffic routing is done.
+
+##### Core
+
+- :fireworks: Routes now have a `name` field [#3764](https://github.com/Kong/kong/pull/3764)
+- :fireworks: Implemented TTL support in the new DAO & Admin API. In particular,
+  PostgreSQL gets a new and more efficient TTL implementation [#3603](https://github.com/Kong/kong/pull/3603),
+  [#3638](https://github.com/Kong/kong/pull/3638)
+- :fireworks: improved performance in router rebuild by decreasing amount of database access
```

Comment:
im not sure this is accurate; #3782 reduces the number of database accesses required in _high concurrency_ situations, but it does not improve the performance of the router rebuild itself.